pandas/tests/tslibs/test_parsing.py

```diff
@@ -25,6 +25,8 @@ def test_parse_time_string(self):
         assert parsed == parsed_lower
         assert reso == reso_lower
 
+        pytest.raises(TypeError, parse_time_string, Period('2018-11-01', 'B'))
```

Comment:
use a context manger here,  make this a separate test, and add the issue as a comment